crypto/fipsmodule/cpucap/cpu_aarch64_linux.c

```diff
@@ -114,6 +123,15 @@ void OPENSSL_cpuid_setup(void) {
   if (hwcap & kSHA512) {
     OPENSSL_armcap_P |= ARMV8_SHA512;
   }
+  if (hwcap & kSHA3) {
+    OPENSSL_armcap_P |= ARMV8_SHA3;
+  }
+
+  OPENSSL_arm_midr = armv8_cpuid_probe();
+  if ((MIDR_IS_CPU_MODEL(OPENSSL_arm_midr, ARM_CPU_IMP_ARM, ARM_CPU_PART_V1)) &&
+      (OPENSSL_armcap_P & ARMV8_SHA3)) {
+    OPENSSL_armcap_P |= ARMV8_AES_GCM_UNROLL8;
+  }
```

Comment:
can we add a comment explaining what is this code doing?

Reply:
Done